### PR TITLE
Release v0.4.0

### PR DESCRIPTION
**Pull request by astoycos**
All the setup needed for cutting our 0.4.0 release 🍾 

- [x] Release notes/Changelog
- [x] crate automation update
- [x] examples release
.....

### CODE REVIEW

**Review by Billy99 — COMMENTED** [review]
In RELEASE.md, all sub-bullets need two additional spaces so they are indented properly.

**Comment by astoycos** [comment]
fixes https://github.com/bpfman/bpfman/issues/1056

File: bpfman-operator/hack/build-release-yamls.sh [hunk 1]
```diff
@@ -33,54 +33,54 @@ KUSTOMIZE=${LOCALBIN}/kustomize
...
 ### TC
-${KUSTOMIZE} build ../examples/config/v${VERSION}/go-tc-counter > release/go-tc-counter-install-v${VERSION}.yaml
-echo "Generated:" go-tc-counter-install-v${VERSION}.yaml
+${KUSTOMIZE} build ../examples/config/v${VERSION}/go-tc-counter > release-v${VERSION}/go-tc-counter-install.yaml
+echo "Generated:" release-v${VERSION}/go-tc-counter-install.yaml
 ### TRACEPOINT
-${KUSTOMIZE} build ../examples/config/v${VERSION}/go-tracepoint-counter > release/go-tracepoint-counter-install-v${VERSION}.yaml
-echo "Generated:" go-tracepoint-counter-install-v${VERSION}.yaml
+${KUSTOMIZE} build ../examples/config/v${VERSION}/go-tracepoint-counter > release-v${VERSION}/go-tracepoint-counter-install.yaml
+echo "Generated:" release-v${VERSION}/go-tracepoint-counter-install.yaml
```
Comment (Billy99) [inline bpfman-operator/hack/build-release-yamls.sh:75]: Need to add kprobe and uprobe, might need go-target.

File: bpfman-operator/hack/build-release-yamls.sh [hunk 2]
```diff
@@ -33,54 +33,54 @@ KUSTOMIZE=${LOCALBIN}/kustomize
...
 ### TC
-${KUSTOMIZE} build ../examples/config/v${VERSION}-ocp/go-tc-counter > release/go-tc-counter-install-ocp-v${VERSION}.yaml
-echo "Generated:" go-tc-counter-install-ocp-v${VERSION}.yaml
+${KUSTOMIZE} build ../examples/config/v${VERSION}-ocp/go-tc-counter > release-v${VERSION}/go-tc-counter-install-ocp.yaml
+echo "Generated:" release-v${VERSION}/go-tc-counter-install-ocp.yaml
 ### TRACEPOINT
-${KUSTOMIZE} build ../examples/config/v${VERSION}-ocp/go-tracepoint-counter > release/go-tracepoint-counter-install-ocp-v${VERSION}.yaml
-echo "Generated:" go-tracepoint-counter-install-ocp-v${VERSION}.yaml
+${KUSTOMIZE} build ../examples/config/v${VERSION}-ocp/go-tracepoint-counter > release-v${VERSION}/go-tracepoint-counter-install-ocp.yaml
+echo "Generated:" release-v${VERSION}/go-tracepoint-counter-install-ocp.yaml
```
Comment (Billy99) [inline bpfman-operator/hack/build-release-yamls.sh:86]: Need to add kprobe and uprobe, might need go-target.

File: changelogs/CHANGELOG-v0.4.0.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
+
+The v0.4.0 release is a minor release which fully transitions bpfman to be a
```
Comment (Billy99) [inline changelogs/CHANGELOG-v0.4.0.md:2]: Should we mention the rebranding, bpfd -> bpfman?

File: changelogs/CHANGELOG-v0.4.0.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+- Support for Uprobe Programs
+- The ability to attach Uprobes inside containers (locally and in Kubernetes), see [the blog](https://bpfman.io/main/blog/2024/02/26/technical-challenges-for-attaching-ebpf-programs-in-containers/) for more.
+- Support for Kprobe Programs
+- Support for Fentry Programs
+- Support for Fexit Programs
+
+Additionally this release provides some new binary crates, the `bpf-metrics-exporter`,
+and the `bpf-log-exporter`, which allow users to gather useful information regarding
+the bpf subsystem.
```
Comment (Billy99) [inline changelogs/CHANGELOG-v0.4.0.md:18]: Do we need to mention the bpfman-rpc binary (it's new for this release)?

File: bpfman-operator/hack/build-release-yamls.sh (hunk 1 above)
Comment (astoycos) [reply bpfman-operator/hack/build-release-yamls.sh:75]: Good catch I think I got it!

File: changelogs/CHANGELOG-v0.4.0.md (hunk 3 above)
Comment (astoycos) [reply changelogs/CHANGELOG-v0.4.0.md:2]: Hrm yeah I guess I should :)

File: changelogs/CHANGELOG-v0.4.0.md (hunk 4 above)
Comment (astoycos) [reply changelogs/CHANGELOG-v0.4.0.md:18]: Yeah I can give it a call out!

File: bpfman-operator/config/bpfman-operator-deployment/kustomization.yaml [hunk 5]
```diff
@@ -5,4 +5,4 @@ kind: Kustomization
 images:
 - name: quay.io/bpfman/bpfman-operator
   newName: quay.io/bpfman/bpfman-operator
-  newTag: latest
+  newTag: v0.4.0
```
Comment (Billy99) [inline bpfman-operator/config/bpfman-operator-deployment/kustomization.yaml:8]: Why are we doing this? Will this need to be backed out once the release is tagged?

Comment (astoycos) [reply bpfman-operator/config/bpfman-operator-deployment/kustomization.yaml:8]: This was an accident good catch!

**Pull request merged**